syntax = "proto3";

package transport_catalogue_serialize;

import "map_renderer.proto";
import "transport_router.proto";

message Coordinate {
    double lat = 1;
    double lon = 2;
}

message StopIdAndDistancePair {
    uint32 key = 1;
    double value = 2;
}

message Stop {
    string name = 1;
    Coordinate coords = 2;
    repeated StopIdAndDistancePair another_stop_id_to_distance = 3;
}

message Bus {
    string name = 1;
    repeated uint32 stop_id = 2;
    bool is_circular_route = 3;
}

message TransportCatalogue {
    repeated Stop stop = 1;
    repeated Bus bus = 2;
    RendererSettings renderer_settings = 3;
    Router router = 4;
}